machine/v1beta1/types_provider.go

```diff
@@ -124,6 +124,23 @@ const (
 // ConditionType is a valid value for Condition.Type.
 type ConditionType string
 
+// Valid conditions for a machine.
+const (
+	// MachineCreated indicates whether the machine has been created or not. If not,
+	// it should include a reason and message for the failure.
+	MachineCreation ConditionType = "MachineCreation"
```

Comment:
I think we should add a note that this is used on certain platforms and it's preferred to use the other condition type `MachineCreated`